### PR TITLE
Adding powermeter interface to DcSupplySimulator

### PR DESCRIPTION
**Pull request by SebaLukas**
## Describe your changes
see title

## Issue ticket number and link
Missing powermeter interface in DcSupplySimulator

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [ ] I have made corresponding changes to the documentation
- [x] I read the [contribution documentation](https://github.com/EVerest/EVerest/blob/main/CONTRIBUTING.md) and made sure that my changes meet its requirements

### CODE REVIEW

**Review by hikinggrass — APPROVED** [review]

**Review by barsnick — COMMENTED** [review]
Nice. Very welcomed.

In the original PR for DcSupplySimulator, the wish was expressed to obsolete the JsDcSupplySimulator in the same step. Perhaps this should be done here.

**Comment by barsnick** [comment]
Oops, too late, already merged. 😁

File: modules/simulation/DCSupplySimulator/main/power_supply_DCImpl.cpp [hunk 1]
```diff
@@ -93,32 +95,56 @@ void power_supply_DCImpl::handle_setImportVoltageCurrent(double& voltage, double
...
+    types::powermeter::Powermeter powermeter;
+
+    powermeter.timestamp = Everest::Date::to_rfc3339(date::utc_clock::now());
+    powermeter.meter_id = "DC_POWERMETER";
+
+    if (connector_current > 0) {
+        energy_import_total += (connector_voltage * connector_current * 0.5) / 3600;
+    }
+    if (connector_current < 0) {
+        energy_export_total += (connector_voltage * -connector_current * 0.5) / 3600;
```
[the remark this answers is not in the document]
Comment (SebaLukas) [reply modules/simulation/DCSupplySimulator/main/power_supply_DCImpl.cpp:118]: Good point :+1: I'll change that quickly

File: modules/simulation/DCSupplySimulator/powermeter/powermeterImpl.cpp [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+powermeterImpl::handle_start_transaction(types::powermeter::TransactionReq& value) {
+    return {types::powermeter::TransactionRequestStatus::OK};
+}
+
+types::powermeter::TransactionStopResponse powermeterImpl::handle_stop_transaction(std::string& transaction_id) {
+    return {types::powermeter::TransactionRequestStatus::NOT_SUPPORTED,
+            {},
+            {},
+            "DcSupplySimulator does not support stop transaction request."};
+}
```
[the remark this answers is not in the document]
Comment (SebaLukas) [reply modules/simulation/DCSupplySimulator/powermeter/powermeterImpl.cpp:25]: I took this directly from the JsDcSupplySimulator module. But it makes sense that the StartTransaction is not supported here either. I will change that too.

**Pull request merged**